Notification URL with localhost does not work

Using a noticification URL for an organisation containing "localhost" causes an ssl exception
localhost != <<<hostname>>>

Logging catalina.out:

09:12:48.465 [wsSmartMeteringResponsesMessageListenerContainer-1] ERROR (NotificationService.java:62) c.a.o.a.w.s.a.s.NotificationService - Add device exception
org.springframework.ws.client.WebServiceIOException: I/O error: hostname in certificate didn't match: <localhost> != <osgp-tst.cloudapp.net>; nested exception is javax.net.ssl.SSLException: hostname in certificate didn't match: <localhost> != <osgp-tst.cloudapp.net>
        at org.springframework.ws.client.core.WebServiceTemplate.sendAndReceive(WebServiceTemplate.java:561) ~[spring-ws-core-2.2.0.RELEASE.jar:2.2.0.RELEASE]
        at org.springframework.ws.client.core.WebServiceTemplate.marshalSendAndReceive(WebServiceTemplate.java:390) ~[spring-ws-core-2.2.0.RELEASE.jar:2.2.0.RELEASE]
        at org.springframework.ws.client.core.WebServiceTemplate.marshalSendAndReceive(WebServiceTemplate.java:383) ~[spring-ws-core-2.2.0.RELEASE.jar:2.2.0.RELEASE]
        at org.springframework.ws.client.core.WebServiceTemplate.marshalSendAndReceive(WebServiceTemplate.java:373) ~[spring-ws-core-2.2.0.RELEASE.jar:2.2.0.RELEASE]
        at com.alliander.osgp.adapter.ws.smartmetering.infra.ws.SendNotificationServiceClient.sendNotification(SendNotificationServiceClient.java:44) ~[SendNotificationServiceClient.class:na]
        at com.alliander.osgp.adapter.ws.smartmetering.application.services.NotificationService.sendNotification(NotificationService.java:59) ~[NotificationService.class:na]
        at com.alliander.osgp.adapter.ws.smartmetering.application.services.NotificationService$$FastClassBySpringCGLIB$$4f9b1f15.invoke(<generated>) [spring-core-4.2.0.RELEASE.jar:na]
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204) [spring-core-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:717) [spring-aop-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157) [spring-aop-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.validation.beanvalidation.MethodValidationInterceptor$HibernateValidatorDelegate.invokeWithinValidation(MethodValidationInterceptor.java:173) [spring-context-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.validation.beanvalidation.MethodValidationInterceptor.invoke(MethodValidationInterceptor.java:132) [spring-context-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) [spring-aop-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99) [spring-tx-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:281) [spring-tx-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96) [spring-tx-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) [spring-aop-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:653) [spring-aop-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at com.alliander.osgp.adapter.ws.smartmetering.application.services.NotificationService$$EnhancerBySpringCGLIB$$75fdc5f.sendNotification(<generated>) [spring-core-4.2.0.RELEASE.jar:na]
        at com.alliander.osgp.adapter.ws.smartmetering.infra.jms.SmartMeteringResponseMessageListener.onMessage(SmartMeteringResponseMessageListener.java:52) [SmartMeteringResponseMessageListener.class:na]
        at org.springframework.jms.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:746) [spring-jms-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.jms.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:684) [spring-jms-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.jms.listener.AbstractMessageListenerContainer.doExecuteListener(AbstractMessageListenerContainer.java:651) [spring-jms-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.doReceiveAndExecute(AbstractPollingMessageListenerContainer.java:315) [spring-jms-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.receiveAndExecute(AbstractPollingMessageListenerContainer.java:253) [spring-jms-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.invokeListener(DefaultMessageListenerContainer.java:1150) [spring-jms-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.executeOngoingLoop(DefaultMessageListenerContainer.java:1142) [spring-jms-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.run(DefaultMessageListenerContainer.java:1039) [spring-jms-4.2.0.RELEASE.jar:4.2.0.RELEASE]
        at java.lang.Thread.run(Thread.java:744) [na:1.7.0_55]
Caused by: javax.net.ssl.SSLException: hostname in certificate didn't match: <localhost> != <osgp-tst.cloudapp.net>
        at org.apache.http.conn.ssl.AbstractVerifier.verify(AbstractVerifier.java:238) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.conn.ssl.BrowserCompatHostnameVerifier.verify(BrowserCompatHostnameVerifier.java:54) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.conn.ssl.AbstractVerifier.verify(AbstractVerifier.java:159) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.conn.ssl.AbstractVerifier.verify(AbstractVerifier.java:140) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.conn.ssl.SSLSocketFactory.verifyHostname(SSLSocketFactory.java:561) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:536) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:403) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:177) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:304) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:611) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:446) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:863) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:106) ~[httpclient-4.3.5.jar:4.3.5]
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:57) ~[httpclient-4.3.5.jar:4.3.5]
        at org.springframework.ws.transport.http.HttpComponentsConnection.onSendAfterWrite(HttpComponentsConnection.java:120) ~[spring-ws-core-2.2.0.RELEASE.jar:2.2.0.RELEASE]
        at org.springframework.ws.transport.AbstractWebServiceConnection.send(AbstractWebServiceConnection.java:48) ~[spring-ws-core-2.2.0.RELEASE.jar:2.2.0.RELEASE]
        at org.springframework.ws.client.core.WebServiceTemplate.sendRequest(WebServiceTemplate.java:654) ~[spring-ws-core-2.2.0.RELEASE.jar:2.2.0.RELEASE]
        at org.springframework.ws.client.core.WebServiceTemplate.doSendAndReceive(WebServiceTemplate.java:603) ~[spring-ws-core-2.2.0.RELEASE.jar:2.2.0.RELEASE]
        at org.springframework.ws.client.core.WebServiceTemplate.sendAndReceive(WebServiceTemplate.java:555) ~[spring-ws-core-2.2.0.RELEASE.jar:2.2.0.RELEASE]
        ... 28 common frames omitted

--- a/osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/config/WebServiceConfig.java
+++ b/osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/config/WebServiceConfig.java
@@ -65,6 +65,8 @@ public class WebServiceConfig {
     private static final String PROPERTY_NAME_WEBSERVICE_KEYSTORE_PASSWORD = "web.service.keystore.password";
     private static final String PROPERTY_NAME_WEBSERVICE_KEYSTORE_TYPE = "web.service.keystore.type";
 
+    private static final String PROPERTY_NAME_WEBSERVICE_NOTIFICATION_URL = "web.service.notification.url";
+
     private static final String PROPERTY_NAME_MARSHALLER_CONTEXT_PATH_SMARTMETERING_NOTIFICATION = "jaxb2.marshaller.context.path.smartmetering.notification";
 
     private static final Logger LOGGER = LoggerFactory.getLogger(WebServiceConfig.class);
@@ -118,6 +120,11 @@ public class WebServiceConfig {
     @Bean
     public NotificationMapper notificationMapper() {
         return new NotificationMapper();
+    }
+
+    @Bean
+    public String notificationURL() {
+        return this.environment.getRequiredProperty(PROPERTY_NAME_WEBSERVICE_NOTIFICATION_URL);
     }
 
     // Client WS code

--- a/osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/services/NotificationService.java
+++ b/osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/services/NotificationService.java
@@ -17,7 +17,6 @@ import org.springframework.validation.annotation.Validated;
 import com.alliander.osgp.adapter.ws.schema.smartmetering.notification.Notification;
 import com.alliander.osgp.adapter.ws.schema.smartmetering.notification.NotificationType;
 import com.alliander.osgp.adapter.ws.smartmetering.infra.ws.SendNotificationServiceClient;
-import com.alliander.osgp.domain.core.entities.Organisation;
 import com.alliander.osgp.domain.core.repositories.OrganisationRepository;
 import com.alliander.osgp.domain.core.validation.Identification;
 import com.alliander.osgp.shared.exceptionhandling.FunctionalException;
@@ -35,6 +34,9 @@ public class NotificationService {
     @Autowired
     private OrganisationRepository organisationRepository;
 
+    @Autowired
+    private String notificationURL;
+
     public NotificationService() {
         // Parameterless constructor required for transactions
     }
@@ -51,13 +53,9 @@ public class NotificationService {
         notification.setCorrelationUid(correlationUid);
         notification.setNotificationType(notificationType);
 
-        final Organisation organisation = this.organisationRepository
-                .findByOrganisationIdentification(organisationIdentification);
-        final String notificationURL = organisation.getNotificationURL();
-
         try {
             this.sendNotificationServiceClient.sendNotification(organisationIdentification, notification,
-                    notificationURL);
+                    this.notificationURL);
         } catch (final Exception e) {
             LOGGER.error("Add device exception", e);
         }

--- a/osgp-core/src/main/resources/db/migration/V2015.013__Removed organisation_notification_url.sql
+++ b/osgp-core/src/main/resources/db/migration/V2015.013__Removed organisation_notification_url.sql
@@ -1,0 +1,1 @@
+ALTER TABLE organisation DROP COLUMN notificationurl;

--- a/osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/entities/Organisation.java
+++ b/osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/entities/Organisation.java
@@ -65,9 +65,6 @@ public class Organisation extends AbstractEntity {
     @Column(nullable = false)
     private boolean enabled;
 
-    @Column(nullable = false)
-    private String notificationURL;
-
     /**
      * Comma separated list of domains organisation is allowed to access.
      */
@@ -202,10 +199,6 @@ public class Organisation extends AbstractEntity {
 
     public void setIsEnabled(final boolean value) {
         this.enabled = value;
-    }
-
-    public String getNotificationURL() {
-        return this.notificationURL;
     }
 
     public void setDomains(final List<PlatformDomain> domains) {

